### PR TITLE
Update README for v2

### PR DESCRIPTION
**Pull request by rtfb**
Properly document v2. This should also get ported to master.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,23 +16,44 @@ It started as a translation from C of [Sundown][3].
...
+* Cleaned up API
+* A separate call to [`Parse`][4], which produces an abstract syntax tree for
+  the document
+* Latest bug fixes
+* Flexibility to easily add your own rendering extensions
+
+Potential drawbacks:
+
+* Our benchmarks show v2 to be slightly slower than v1. Currently in the
+  ballpark of around 15%.
```
Comment (shurcooL) [inline README.md]: Is this before or after #322? With many of the optimizations such as removing the preprocess step, using `io.Writer` and predeclared `[]byte` slices, what's causing v2 to still be slower?

File: README.md [hunk 2]
```diff
@@ -256,3 +276,6 @@ License
    [1]: http://daringfireball.net/projects/markdown/ "Markdown"
    [2]: http://golang.org/ "Go Language"
    [3]: https://github.com/vmg/sundown "Sundown"
+   [4]: https://godoc.org/gopkg.in/russross/blackfriday.v2#Parse "Parse func"
+   [5]: https://github.com/microcosm-cc/bluemonday "Bluemonday"
+   [6]: http://labix.org/gopkg.in "gopkg.in"
```
Comment (shurcooL) [inline README.md]: > http://labix.org/gopkg.in "gopkg.in"

Use `https://` please.

File: README.md [hunk 3]
```diff
@@ -256,3 +276,6 @@ License
    [1]: http://daringfireball.net/projects/markdown/ "Markdown"
    [2]: http://golang.org/ "Go Language"
```
Comment (shurcooL) [inline README.md]: Might as well update these 2 old links to use https too. They both support it.

File: README.md (hunk 1 above)
Comment (rtfb) [inline README.md]: After. Before that, it was around 2.5x slower. The remaining slowdown is very washed out, couldn't pick it out yet.

**Pull request merged**